cuda_core/cuda/core/experimental/_event.py

```diff
@@ -111,6 +113,8 @@ def _init(cls, options: Optional[EventOptions] = None):
         if options.support_ipc:
             raise NotImplementedError("WIP: https://github.com/NVIDIA/cuda-python/issues/103")
         self._mnff.handle = handle_return(driver.cuEventCreate(flags))
+        self._device_id = int(handle_return(driver.cuCtxGetDevice()))
+        self._ctx_handle = handle_return(driver.cuStreamGetCtx(self._mnff.handle))
```

Comment:
Let's avoid driver calls if possible to reduce overhead. The same info can be passed in from within the two public constructors (I think!) `Device.create_event` and `Stream.record`.